docs/refguide/api.md

```diff
@@ -1,46 +1,129 @@
+
+Creates a new observable array based on the provided value. Use `shallowArray` if the values in the array should not be turned into observables.
+
+[&laquo;`details`&raquo;](array.md)
+
+### `observable.map(value)` & `observable.shallowMap(value)`
+
+Creates a new observable map based on the provided value. Use `shallowMap` if the values in the array should not be turned into observables.
+Use `map` whenever you want to create a dynamically keyed collections and the addition / removal of keys needs to be observed.
+Note that only string keys are supported.
+
+[&laquo;`details`&raquo;](map.md)
+
+### `extendObservable` & `extendShallowObservable`
+Usage: `extendObservable(target, ...propertyMaps)`. For each key/value pair in each `propertyMap` a (new) observable property will be introduced on the target object.
 This can be used in constructor functions to introduce observable properties without using decorators.
-If a value of the `propertyMap` is an argumentless function, a *computed* property will be introduced.
+If a value of the `propertyMap` is a getter function, a *computed* property will be introduced.
+
+Use `extendShallowObservable` if the new properties should not be infective (that is; newly assigned values should not be turned into observables automatically).
+Note that `extendObservable` enhances existing objects, unlike `observable.object` which creates a new object.
+
 [&laquo;details&raquo;](extend-observable.md)
 
+### Modifiers
+
+Modifiers can be used decorator or in combination with `extendObservable` and `observable.object` to change the autoconversion rules for specific properties.
+
+The following modifiers are available:
+
+* `observable.deep`: This is the default modifier, used by any observable. It converts any assigned, non-primitive value into an observable if it isn't one yet.
+* `observable.ref`: Disables automatic observable conversion, just creates an observable reference instead.
+* `observable.shallow`: Can only used in combination with collections. Turns any assigned collection into an collection, which is shallowly observable (instead of deep). In other words; the values inside the collection won't become observables automatically.
+* `computed`: Creates a derived property, see [`computed`](computed-decorator.md)
+* `action`: Creates an action, see [`action`](action.md)
+
+Modifiers can be used as decorator:
+
+```javascript
+class TaskStore {
+    @observable.shallow tasks = []
+}
+```
+
+Or as property modifier in combination with `observable.object` / `observable.extendObservable`.
+Note that modifiers always 'stick' to the property. So they will remain in effect even if a new value is assigned.
+
+```javascript
+const taskStore = observable({
+    tasks: observable.shallow([])
+})
+```
+
+[&laquo;details&raquo;](modifiers.md)
+
+
 ## Computed values
 
 Usage:
 * `computed(() => expression)`
+* `computed(() => expression, (newValue) => void)`
```

Comment:
`computed(() => expression, (newValue) => void)` isn't explained in the `computed` docs, only `computed(() => expression)` and `computed(() => expression, options)` are explained